cfpb/cfgov-refresh
# Replace references to part.effective_version when rendering regulation sections

This PR will make it possible to view drafts of effective versions when no non-draft version exists.

Before, we relied on the `part.effective_version` property in several places, which explicitly queries for non-draft versions and returns `None` when no non-draft version exists. This precludes viewing a draft effective version when no non-draft exists.

Attempting to view a draft without permission now raises a 404 rather than PermissionDenied. This seems sensible to me as well.

## Testing

1. Navigate through iRegs as normal (http://localhost:8000/policy-compliance/rulemaking/regulations/) and see that everything works
2. Edit a regulation with a single effective version, like D (http://localhost:8000/admin/regulations3k/effectiveversion/?part=3) and check the "Draft" checkbox to make that effective version a draft.
3. Navigate to the sections of that regulation (http://localhost:8000/admin/regulations3k/section/?subpart=436) and click the "View Draft" button and verify that you can view the draft normally.
4. Open a private browser window (or logout) and attempt to give that regulation again (http://localhost:8000/policy-compliance/rulemaking/regulations/1004/2011-07-22/1/) and verify that you see a 404.

## Checklist
